tests/publiccloud/imageversion_check.pm

```diff
@@ -0,0 +1,129 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: This test run checks if the used image version is up-to-date
+#
+# Maintainer: Felix Niederwanger <felix.niederwanger@suse.de>
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use version_utils;
+use WWW::Mechanize;
+
+# Fetch links from a given url that match the given regexp
+sub fetch_matching_links {
+    my $url   = $_[0];
+    my $match = $_[1];
+    my $mech  = WWW::Mechanize->new();
+    $mech->get($url);
+    my @links = map { $_->url } $mech->links();
+    return grep { $_ =~ $match } @links;
+}
+
+# Compare two version strings (e.g. '0.9.1' and '0.9.0' or '0.141' and '0.139')
+# returns -1 if the first is smaller, 1 if larger and 0 if they are the same
+sub version_cmp {
+    my @v1 = split('\.', $_[0]);
+    my @v2 = split('\.', $_[1]);
+
+    my $l = scalar @v1;
+    die "No versions to compare"  if ($l <= 0);
+    die "Version scheme mismatch" if ($l != scalar @v2);
+    for (my $i = 0; $i < $l; $i++) {
+        if ($v1[$i] < $v2[$i]) {
+            return -1;
+        } elsif ($v1[$i] > $v2[$i]) {
+            return 1;
+        }
+    }
+    return 0;
+
+}
+
+# Checks if a given image link is up-to-date
+sub get_latest_version {
+    my $image    = $_[0];
```

Comment:
yeah your version will work but this is common practice in this repo :
```suggestion
    my ($image)    = @_;
```

Reply:
Updated accordingly